src/Elliptic/Systems/Poisson/FirstOrderSystem.hpp

```diff
@@ -33,52 +33,77 @@ namespace Poisson {
  *
- * \details This system formulates the Poisson equation \f$-\Delta u(x) =
- * f(x)\f$ as the set of coupled first-order PDEs
+ * \details This system formulates the Poisson equation \f$-\Delta_\gamma u(x) =
+ * f(x)\f$ on a background metric \f$\gamma\f$ as the set of coupled first-order
+ * PDEs
+ *
+ * \f[
+ * -\frac{1}{\sqrt{\gamma}} \partial_i \sqrt{\gamma}\gamma^{ij} v_j(x) = f(x) \\
+ * -\partial_i u(x) + v_i(x) = 0
+ * \f]
+ *
+ * where we have chosen the field gradient as an auxiliary variable \f$v_i\f$.
+ * This scheme also goes by the name of _mixed_ or _flux_ formulation (see e.g.
+ * \cite Arnold2002). The reason for the latter name is that we can write the
+ * set of coupled first-order PDEs in flux-form
+ *
  * \f[
- * -\nabla \cdot \boldsymbol{v}(x) = f(x) \\
- * \nabla u(x) - \boldsymbol{v}(x) = 0
+ * -\partial_i F^i_A + S_A = f_A(x)
  * \f]
- * where we make use of an auxiliary variable \f$\boldsymbol{v}\f$. This scheme
- * also goes by the name of _mixed_ or _flux_ formulation (see e.g.
- * \cite Arnold2002). The auxiliary variable is treated on the same footing as
- * the field \f$u\f$. This allows us to make use of the DG architecture
- * developed for coupled first-order PDEs, in particular the flux communication
- * and lifting code. It does, however, introduce auxiliary degrees of freedom
- * that can be avoided in the _primal formulation_. Furthermore, the linear
- * operator that represents the DG discretization for this system is not
- * symmetric (since no mass operator is applied) and has both positive and
- * negative eigenvalues. These properties further increase the computational
- * cost (see \ref LinearSolverGroup) and are remedied in the primal formulation.
+ *
+ * by choosing the fluxes and sources in terms of the system variables
+ * \f$u(x)\f$ and $v_i(x)}\f$ as
+ *
+ * \f{align}
+ * F^i_u &= \sqrt{\gamma}\gamma^{ij} v_j(x) \quad F^i_{v_j} &= u \delta^i_j \\
+ * S_u &= 0 \quad S_{v_i} &= v_j
+ * f_u &= \sqrt{\gamma} f(x) \quad f_{v_i} &= 0 \text{.}
+ * \f}
+ *
+ * Note that we use the system variables to index the fluxes and sources, which
+ * we also do in the code by using DataBox tags.
+ * Also note that we have defined the _fixed sources_ \f$f_A\f$ as those source
+ * terms that are independent of the system variables.
+ *
+ * The field gradient \f$v_i\f$ is treated on the same footing as the field
+ * \f$u\f$ in this first-order formulation. This allows us to make use of the DG
+ * architecture developed for coupled first-order hyperbolic PDEs in flux-form,
+ * in particular the flux communication and lifting code. It does, however,
+ * introduce auxiliary degrees of freedom that can be avoided in the
+ * second-order (or _primal_) formulation. Furthermore, the linear operator that
+ * represents the DG discretization for this system is not symmetric. This
+ * property further increase the computational cost (see \ref LinearSolverGroup)
+ * and is remedied in the second-order formulation.
```

Comment:
Will preconditioning be more complicated for this first-order formulation?

Reply:
No, preconditioning doesn't care about the formulation :) It's an operation on the linear operator and agnostic to how it's built.